### PR TITLE
New feature: detect and fill in data streams for docs() query

### PR DESCRIPTION
**Pull request by romanchyla**

### CODE REVIEW

File: solr/views.py [hunk 1]
```diff
@@ -4,6 +4,13 @@
 import json
 from models import Limits
 from sqlalchemy import or_
+from werkzeug import MultiDict
+try:
+    from cStringIO import StringIO
+except:
+    from StringIO import StringIO
```
Comment (kelockhart) [inline solr/views.py]: Why this try/except? Will cStringIO sometimes not be available?

File: solr/views.py [hunk 2]
```diff
@@ -178,6 +185,93 @@ def _get_host(self, url):
...
+                    for x in v:
+                        if 'docs(' in x:
+                            streams.update(self._extract_docs_values(x))
+                
+        if len(streams) == 0:
+            return
+        
+        # must verify the input is there
+        for sn in streams:
+            if sn in params: # it can be in the parameters, which is OK...
```
Comment (kelockhart) [inline solr/views.py]: When would the docs be in the params?

File: solr/views.py (hunk 1 above)
Comment (romanchyla) [reply solr/views.py]: in older python insllations; i guess this is just an old habit

File: solr/views.py (hunk 2 above)
Comment (romanchyla) [reply solr/views.py]: if one decided to save params not in the bigquery (stream) but inside url params; that's the only case when docs() can be used against /search/query endpoint

**Pull request merged**